code/modules/mapping/mapping_helpers.dm

```diff
@@ -348,12 +528,31 @@
 	pipe.update_layer()
 	pipe.paint(pipe_color)
 
-//	var/obj/machinery/door/airlock/airlock = locate(/obj/machinery/door/airlock) in loc
-//	if(airlock)
-//		airlock.unres_sides ^= dir
-//	else
-//		log_world("### MAP WARNING, [src] failed to find an airlock at [AREACOORD(src)]")
+/obj/effect/mapping_helpers/airlock/unres
+	name = "airlock unrestricted side helper"
+	icon_state = "airlock_unres_helper"
+
+/obj/effect/mapping_helpers/airlock/unres/Initialize(mapload)
+	. = ..()
+	var/obj/machinery/door/airlock/airlock = locate(/obj/machinery/door/airlock) in loc
+	if(!airlock)
+		stack_trace("### MAP WARNING, [src] failed to find an airlock at [AREACOORD(src)]")
+		return
```

Comment:
This could just be a CRASH("### MAP WARNING, [src] failed to find an airlock at [AREACOORD(src)]") instead of stack tracing and returning